src/utils/numberFormatter.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2022 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+export interface NumberFormatterOptions {
+  truncate?: boolean
+}
+
+export const numberFormatter: (value?: number, options?: NumberFormatterOptions) => string = (
```

Comment:
I didn't go over the internals of this function but I would check if the built-in library [Intl.NumberFormat](https://developer.mozilla.org/en-US/docs/web/javascript/reference/global_objects/intl/numberformat) is enough to handle what we need.

e.g - this snippet result is `654.32K`:
```
new Intl.NumberFormat(
        'default',
        { notation: "compact" , compactDisplay: "short", unitDisplay: "long", minimumFractionDigits: 0, maximumFractionDigits: 2 }
    ).format(654321);```

Reply:
sure will check this, thanks for suggesting it.